docs/content/integrations/openai.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: "OpenAI + Dagster"
+description: The `dagster-openai` library provides the ability to build OpenAI pipelines with Dagster and log OpenAI API usage metadata in Dagster Insights.
+---
+
+# OpenAI + Dagster
+
+The `dagster-openai` library allows you to build OpenAI pipelines with Dagster and log OpenAI API usage metadata in [Dagster Insights](/dagster-cloud/insights).
+
+Using this library's <PyObject module="dagster_openai" object="OpenAIResource" />, you can easily interact with the [OpenAI REST API](https://platform.openai.com/docs/introduction) via the [OpenAI Python API](https://github.com/openai/openai-python).
+
+When used with Dagster's [Software-defined Assets](/concepts/assets/software-defined-assets), the resource automatically logs OpenAI usage metadata in asset metadata. See the [Relevant APIs](#relevant-apis) section for more information.
+
+---
+
+## Getting started
+
+Before you get started with the `dagster-openai`library, we recommend familiarizing yourself with the [OpenAI Python API library](https://github.com/openai/openai-python), which this integration uses to interact with the [OpenAI REST API](https://platform.openai.com/docs/introduction).
+
+---
+
+## Prerequisites
+
+To get started, install the `dagster` and `dagster-openai` Python packages:
+
+```bash
+pip install dagster dagster-openai
+```
+
+Note that you will need an OpenAI [API key](https://platform.openai.com/api-keys) to use the resource, which can be generated in your OpenAI account.
+
+---
+
+## Connecting to OpenAI
+
+The first step in using OpenAI with Dagster is to tell Dagster how to connect to an OpenAI client using an OpenAI [resource](/concepts/resources). This resource contains the credentials needed to interact with OpenAI API.
+
+We will supply our credentials as environment variables by adding them to a `.env` file. For more information on setting environment variables in a production setting, see [Using environment variables and secrets](/guides/dagster/using-environment-variables-and-secrets).
+
+```bash
+# .env
+
+OPENAI_API_KEY=...
+```
+
+Then, we can instruct Dagster to authorize the OpenAI resource using the environment variables:
+
+```python startafter=start_example endbefore=end_example file=/integrations/openai/resource.py dedent=4
```

Comment:
```suggestion
```python startafter=start_example endbefore=end_example file=/integrations/openai/resource.py
```

`dedent` is used for code that are inside a function so the code block can dedent. in this case, we dont need this. otherwise, it'd look off like it's currently showing:
<img width="778" alt="image" src="https://github.com/dagster-io/dagster/assets/4531914/7d6f649e-9aff-49d9-8c2e-6f8b5c259e37">